src/vg_log_segments.erl

```diff
@@ -220,17 +220,27 @@ find_in_log(_Log, Id, Position, LastSize, {ok, <<FileId:64/signed, _Size:32/sign
     {Position, LastSize};
 find_in_log(_Log, Id, Position, LastSize, {ok, <<Id:64/signed, _Size:32/signed>>}) ->
     {Position+LastSize, 0};
-find_in_log(Log, Id, Position, LastSize, {ok, <<_:64/signed, Size:32/signed>>}) ->
+find_in_log(Log, Id, Position, LastSize, {ok, <<FileId:64/signed, Size:32/signed>>}) ->
     case file:read(Log, Size + 12) of
         {ok, <<_:Size/binary, Data:12/binary>>} ->
             find_in_log(Log, Id, Position+LastSize, Size+12, {ok, Data});
-        {ok, <<_:Size/binary>>} ->
-            {Position+LastSize, Size+12};
+        {ok, <<D:Size/binary>>} ->
+            case D of
+                <<_LeaderEpoch:32/signed-integer,
+                  ?MAGIC_TWO:8/signed-integer,
+                  _CRC:32/signed-integer,
+                  _Attributes:16/signed-integer,
+                  LastOffsetDelta:32/signed-integer, _/binary>> when LastOffsetDelta + FileId >= Id ->
+                    {Position+LastSize, Size+12};
```

Comment:
So, I know it's not directly related to this PR, but could we make 12 a macro here to make it clear what the offset represents to those reading the code?

Reply:
Good point, this has bitten us before actually. I'll make that change.